src/controller/event/operator/operator.go

```diff
@@ -0,0 +1,41 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package operator
+
+import (
+	"context"
+
+	"github.com/goharbor/harbor/src/common/security"
+	"github.com/goharbor/harbor/src/lib/log"
+	"github.com/goharbor/harbor/src/pkg/user"
+)
+
+// FromContext return the event operator from context
+func FromContext(ctx context.Context) string {
```

Comment:
operator.FromContext() is actually return a username, it's hard to understand.

Reply:
Yes, it returns the username of the security context and returns `admin` when the `IsSolutionUser` of security context returns `true`.

This is the original logic to get the operator of `CreateProject` and `DeleteProject` events. The code will be duplicated in the creating and delete the project, so extract it as a function will be better. 

I found that the metadata in `src/controller/event/metadata` almost all use the username of the security context as their operator except `ScanImageMetaData`. In this case, it's fine to provide a function to get the operator from the context.

And some metadata is very strange that they save the context in their struct and fill the operator from the security context when resolving the event. There are `PushArtifactEventMetadata`, `PullArtifactEventMetadata`, `DeleteArtifactEventMetadata`, `ArtifactLabeledMetadata`, `DeleteRepositoryEventMetadata`, `CreateTagEventMetadata` and `DeleteTagEventMetadata`. 

This is a not good practice to save the context from the HTTP request in the notification events because the event sending in another goroutine than the goroutine which processes the HTTP request. What will happen when the HTTP goroutine was finished but the goroutine which sends the event still not be scheduled by the runtime of Golang?